src/countdown-timer/countdown-timer.tsx

```diff
@@ -0,0 +1,150 @@
+import { TimeHelper } from "../util/time-helper";
+import { ClockIcon } from "@lifesg/react-icons";
+import { MediaWidths } from "../spec/media-spec";
+
+type Position = "relative" | "fixed";
+
+export const CountdownTimer = ({
+    className,
+    timer,
+    notifyTimer,
+    offset,
+    mobileOffset,
+    show,
+    "data-testid": testId,
+    onDuration,
+    onNotify,
+    onFinish,
+}: CountdownTimerProps) => {
+    // =============================================================================
+    // CONST, STATE, REF
+    // =============================================================================
+
+    const wrapperRef = useRef<HTMLDivElement>();
+    const isNotified = useRef<boolean>(false);
+    const [offsetY, setOffsetY] = useState<number>(0);
+    const [isPlaying, setIsPlaying] = useState(false);
+
+    const [remainingSeconds] = useTimer(timer, isPlaying);
+    const { ref: stickyRef, inView } = useInView({
+        threshold: 1,
+        rootMargin: `${offsetY * -1}px 0px 0px 0px`,
+    });
+
+    const isMobile = useMediaQuery({
+        maxWidth: MediaWidths.mobileL,
+    });
+
+    // =============================================================================
+    // EFFECTS
+    // =============================================================================
+
+    useEffect(() => {
+        setIsPlaying(show);
+    }, [show]);
+
+    useEffect(() => {
+        if (remainingSeconds === 0) {
+            performOnFinishHandler();
+        } else if (remainingSeconds <= notifyTimer) {
+            performOnDurationHandler();
+            performOnNotifyHandler();
+        }
+    }, [remainingSeconds]);
+
+    useEffect(() => {
+        const y = getOffsetY();
+        setOffsetY(y);
+    }, [isMobile, offset?.top]);
+
+    useEffect(() => {
+        // reset
+        isNotified.current = false;
+    }, [timer, show]);
+
+    // =============================================================================
+    // HELPER FUNCTIONS
+    // =============================================================================
+
+    const performOnDurationHandler = () => {
+        if (onDuration) {
+            onDuration(remainingSeconds);
+        }
+    };
+
+    const performOnNotifyHandler = () => {
+        if (onNotify && !isNotified.current) {
+            isNotified.current = true;
+            onNotify();
+        }
+    };
+
+    const performOnFinishHandler = () => {
+        setIsPlaying(false);
+
+        if (onFinish) onFinish();
+    };
+
+    function getOffsetY() {
+        const desktopTop = offset?.top ?? 168;
+        const mobileTop = mobileOffset?.top ?? 80;
+        const offsetY = isMobile ? mobileTop : desktopTop;
+
+        return offsetY;
+    }
+
+    // =============================================================================
+    // RENDER FUNCTION
+    // =============================================================================
+
+    const renderCountdown = (position: Position) => {
```

Comment:
it might be cleaner to split the rendering into `renderTimer()`, `renderFixedCountdown()`, and `renderCountdown()` so that there's no need to keep checking `position === "relative"`

similarly the styled components can have a `Countdown` and `FixedCountdown` that extend from `BaseCountdown`